modules/alloy/src/main/ts/ephox/alloy/behaviour/sliding/SlidingApis.ts

```diff
@@ -153,6 +153,20 @@ const toggleGrow = (component: AlloyComponent, slideConfig: SlidingConfig, slide
   f(component, slideConfig, slideState);
 };
 
+const immediateGrow = (component: AlloyComponent, slideConfig: SlidingConfig, slideState: SlidingState): void => {
+  if (!slideState.isExpanded()) {
+    setGrown(component, slideConfig);
+    // Force current dimension to begin transition
```

Comment:
This looks a bit odd. We put in in the final state then a few lines below the transition is disabled. Wouldn't that stop the animation directly. The comment here isn't very clear. Is the reflow really needed if the class gets changed in the same animation frame?

Reply:
This is how it's done elsewhere here so I'm assuming @ephox-mogran had a good reason for that. We definitely need to remove the transition classes for the case where it's already in a partial growing/shrinking state. I'm not so convinced about the need for the reflow though.

As for the comment, good point it's not really accurate here. What we're really doing is setting the final size value and then making sure it doesn't transition to get to that state.

Reply:
I too am very curious why the transitions can't be disabled before setting the width. The reflow wouldn't be required at that point.

Reply:
> This is how it's done elsewhere here so I'm assuming @ephox-mogran had a good reason for that.

Trial and error from several years ago, usually.  I'd have to look back through the history. It's possible this was done during apollo as well. Sorry, I can't remember off-hand.

Reply:
Looking at the code with Ben, generally the reason you needed reflow was to trigger an animation, so if this isn't being animated, it may not be necessary. However, the edge cases were always when it was half-way through something.

Reply:
The reflow was usually to force it to register its current size so that it could animate to the next one.

Reply:
IIRC, the reflow was required for situations where the css or classes that had just changed impacted the size of the element and its transition states. So we did a reflow after removing classes or adding classes that would have changed CSS width family values. In this case, because we just want to jump to the immediate size, I don't think any of it should be necessary.

Reply:
Summary. Not sure. Remove it. See what happens.

Probably also the one in immediateShrink. If something is relying on that size being set, it could always call Css.reflow. From looking through history, immediateShrink was added as part of expandable forms from pre-apollo ... so it's been there for ages.

Reply:
So I removed the `reflow` as we don't need to trigger an animation and `disableTransition` is also removed as the result. Just noted that it seems to work fine in the manual test, including in the edge case where it is grown from a halfway shrink

Reply:
@tiny-ben-tran that's not what we said to do... per my above comment `disableTransition` is needed and must be kept. In the above we only discussed removing the reflow, so that's all that should have been removed here.

Reply:
Morgan also mentioned above that if we're removing it here then we should also look at removing the one in `immediateShrink`:
> Summary. Not sure. Remove it. See what happens.
> 
> Probably also the one in immediateShrink